transport/algeneva/stream_dialer_test.go

```diff
@@ -0,0 +1,129 @@
+package algeneva
+
+import (
+	"context"
+	"net"
+	"testing"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	algeneva "github.com/getlantern/lantern-algeneva"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/getlantern/radiance/config"
+)
+
+func TestNewStreamDialer(t *testing.T) {
+	certPem := []byte(`-----BEGIN CERTIFICATE-----
+MIIBhTCCASugAwIBAgIQIRi6zePL6mKjOipn+dNuaTAKBggqhkjOPQQDAjASMRAw
+DgYDVQQKEwdBY21lIENvMB4XDTE3MTAyMDE5NDMwNloXDTE4MTAyMDE5NDMwNlow
+EjEQMA4GA1UEChMHQWNtZSBDbzBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABD0d
+7VNhbWvZLWPuj/RtHFjvtJBEwOkhbN/BnnE8rnZR8+sbwnc/KhCk3FhnpHZnQz7B
+5aETbbIgmuvewdjvSBSjYzBhMA4GA1UdDwEB/wQEAwICpDATBgNVHSUEDDAKBggr
+BgEFBQcDATAPBgNVHRMBAf8EBTADAQH/MCkGA1UdEQQiMCCCDmxvY2FsaG9zdDo1
+NDUzgg4xMjcuMC4wLjE6NTQ1MzAKBggqhkjOPQQDAgNIADBFAiEA2zpJEPQyz6/l
+Wf86aX6PepsntZv2GYlA5UpabfT2EZICICpJ5h/iI+i341gBmLiAFQOyTDT+/wQc
+6MF9+Yw1Yy0t
+-----END CERTIFICATE-----`)
+	protocolConfig := &config.ProxyConnectConfig_ConnectCfgAlgeneva{
+		ConnectCfgAlgeneva: &config.ProxyConnectConfig_AlgenevaConfig{Strategy: "test-strategy"},
+	}
+	tests := []struct {
+		name   string
+		cfg    *config.Config
+		assert func(t *testing.T, got transport.StreamDialer, cfg *config.Config, err error)
+	}{
+		{
+			name: "success",
+			cfg: &config.Config{
+				Addr:           "1.1.1.1",
+				CertPem:        certPem,
+				ProtocolConfig: protocolConfig,
+			},
+			assert: func(t *testing.T, got transport.StreamDialer, cfg *config.Config, err error) {
+
```

Comment:
```suggestion
				assert.NoError(t, err)
				assert.NotNil(t, got)
```

We could also assert if the config opts are the same as the config expects.